app/controllers/comments/delete.js

```diff
@@ -0,0 +1,88 @@
+import { db } from "~/utils/db.server";
+import generateSessionIdHash from "~/utils/backend/crypto";
+import { generateMinMaxDates } from "~/utils/backend/comments";
+import { deleteCommentSchema } from "~/utils/backend/validators/comments";
+import { DEFAULT_ERROR_MESSAGE } from "~/utils/backend/constants";
+import {
+  INVALID_PARAMS_FOR_OPERATION_ERROR_MESSAGE,
+  DELETE_COMMENT_ERROR_MESSAGE,
+} from "~/utils/constants";
+
+export const deleteComment = async (body) => {
+  const { error, value } = deleteCommentSchema.validate(body);
+
+  if (error) {
+    return {
+      errors: [
+        {
+          message: INVALID_PARAMS_FOR_OPERATION_ERROR_MESSAGE,
+          detail: error,
+        },
+      ],
+    };
+  }
+
+  const { commentId: id, accessToken, userEmail } = value;
+
+  const sessionHash = generateSessionIdHash(accessToken, id);
+  const { minDate, maxDate } = generateMinMaxDates();
+
+  const deleteCommentResponse = await db.Comments.deleteMany({
```

Comment:
Does regular .delete method not work? Or is there a reason for deleteMany

Reply:
Yes, similar to another issue we had trying to use multiple params inside the 'where' clause to target a specific record with the Prisma method designed to act on a single record ('update' and in this case 'delete'), we get this error if we try to use another property of the model different than its primary key:

`Argument where of type CommentsWhereUniqueInput needs exactly one argument, but you provided id and OR. Please choose one. Available args: 
    type CommentsWhereUniqueInput {
      id?: Int
    }`

That's why we have to use the 'Many' version of the method since it supports multiple filtering properties in the 'where', and since we are including the comment id in the 'where' as well, only one record will be deleted at most in the operation, the following if clause will never be executed if Prisma works as expected, I just added it for extra precaution:

`return {
    error: {
      message: DELETE_COMMENT_ERROR_MESSAGE,
      detail: 'Multiple comments were deleted, which should not have happened, number of affected comments: ${deleteCommentResponse.count}',
    },
  };`